### DISCUSSION TITLE
A clone of M3 Material docs

### DISCUSSION THREAD

**Discussion by EAzari** [material-components/material-web#4176] [Ideas]
Is it possible clone [M3 Material docs](https://m3.material.io/) and start to put Web Components and their docs in their places, to see which components are developed and which are missing

Also, it will help community to use components better

**Comment by VandeurenGlenn** [comment]
@EAzari 
Made one quick, don't mind the colors, was playing with colorThief
It basically just uses the readme's from the docs
https://vandeurenglenn.github.io/unofficial-material-web-catalog/#!/button

**Reply by vdegenne** [reply]
@VandeurenGlenn Yeah this is just the doc. I think the OP meant to make a catalog with the actual usage of the components.

**Reply by VandeurenGlenn** [reply]
I thought docs were kinda doing this? But in a basic way ofc.
I thought the elements in docs are elements that quite stable already/explain enough to work with them.

**Comment by kavishdevar** [comment]
Maybe check [docs website](https://material-components.github.io/material-web/).. It includes a few components with their usages..

**Reply by e111077** [reply]
This is going to be the official doc source and should be linked out from material.io in the coming months. The components on there are the ones that are fully documented.

Please note, the URL may change in the coming weeks to something _more official_

**Comment by e111077** [comment]
**See update [here](https://github.com/material-components/material-web/discussions/4176#discussioncomment-6997624)**

+1 to what @stiblook said, but I wanted to say, if you'd like to improve the docs website, we could always use some help with PRs in writing up some READMEs in the `docs/` folder. If you'd like to see it rendered in the catalog, follow these instructions:

1. clone the repo:
    ```shell
    git clone git@github.com:material-components/material-web
    ```
2. cd into the catalog directory
    ```shell
    cd material-web/catalog
    ```
3. install deps
    ```shell
    npm ci
    ```
4. run the catalog in dev mode
    ```shell
    npm run dev
    ```
5. Edit docs in /docs root folder and include the commented out frontmatter to render in docs sidebar. You can follow [Checkbox](https://github.com/material-components/material-web/blob/main/docs/components/checkbox.md?plain=1) as an example.

**Reply by vdegenne** [reply]
@e111077 I just wanted to make a reference to this issue https://github.com/google/wireit/issues/790
As much as I wanted to contribute I couldn't run the development environment properly and others could face the same issue.

**Reply by e111077** [reply]
What is your environment? OS, OS version, npm version, node version?

**Reply by vdegenne** [reply]
Right now I'm running Fedora 37, node v18.16.0. But I also get this issue from GitHub Codespace, that's why I think it could be a *common* bug with `wireit` hampering any further progress.

**Reply by e111077** [reply]
I cannot reproduce this on GitHub Codespaces using npm v9.6.3 and node v19.9.0 or on a Mac npm v9.4.0 and node v19.6.0 or node v18.16.0.

Can you see if updating node and npm fix this issue?

**Reply by VandeurenGlenn** [reply]
@e111077 
Things def not working
<img width="874" alt="image" src="https://github.com/material-components/material-web/assets/8685280/d88d8a24-de07-4d93-ace6-eb8dd36e3c96">

**Reply by VandeurenGlenn** [reply]
That is from a codespace, looks fine till you keep it running for some time

**Reply by vdegenne** [reply]
Same error, `node v19.9.0` and `npm v9.6.3` exactly

![image](https://github.com/material-components/material-web/assets/2827383/09e020be-edb3-4743-a835-16389d4f1445)

**Reply by e111077** [reply]
I think that is because your codespace does not have enough memory to run all at once. Try running it again and the cached steps should let it advance

**Reply by e111077** [reply]
Specifically try running npm run build:dev

**Reply by VandeurenGlenn** [reply]
After rerun works

**Reply by vdegenne** [reply]
Not really. Sure it pass the failing stage, but have you tried to modify a file to trigger a change?

**Reply by VandeurenGlenn** [reply]
@vdegenne 
Yup I forgot to trigger changes, after +-3 times it fails

<img width="1277" alt="image" src="https://github.com/material-components/material-web/assets/8685280/92ba88eb-18fd-4fdb-87f6-8cdba408f85e">

@e111077

**Reply by kavishdevar** [reply]
had the same issue.. it crashes after sometime.. i set the memory using the flag `--max-old-space-size`, and got the same error, now.

**Reply by kavishdevar** [reply]
Also, @e111077, I tried adding docs for `textfield`, and render it on the catalog, I can't seem to get the components rendered, do I need to change the imports or something anywhere?

EDIT: nvm, got it, need it edit `catalog/src/ssr.ts`

**Comment by e111077** [comment]
Update. Official material design docs should be updated to link out to "web - MWC"

[See checkbox docs for example](https://m3.material.io/components/checkbox/overview)

<img width="1216" alt="image" src="https://github.com/material-components/material-web/assets/5981958/9090fb5c-b303-47bd-9b20-0d547fbea427">


That should link to our GitHub docs and eventually to material-web.dev when we have completed that site post 1.0 ([here it is for checkbox](https://material-web.dev/components/checkbox/)).

We intend to put up more comprehensive examples on material-web.dev post 1.0 launch as 1.0 we are focusing on code quality and github docs
